web-ui/src/context/AppContext.js

```diff
@@ -55,26 +63,65 @@ const AppContextProvider = (props) => {
     updateUserProfile();
   }, []);
 
+  const date = (months, prevCheckinDate) => {
```

Comment:
If I understand this right (and I very much might not) this will produce an array of numbers (e.g. "[2020, 9, 8]") but the service is expecting an ISO 8601 date (e.g. "2020-09-08").  Does React re-render this somewhere in the background or something?

Reply:
No, you're correct. The service was a bit odd because it said it expected the date in this format '2020-09-08' but then returned the date to me in this format [2020, 9, 8]. It actually accepts either format and since it returns an array, I wanted to try to keep it a bit more uniform.

Reply:
ok, wait...what? let's put in a ticket the service to not respond with that weirdness.